src/dags/dag.py

```diff
@@ -107,6 +181,29 @@ def get_ancestors(functions, targets, include_targets=False):
     return ancestors
 
 
+def harmonize_and_check_functions_targets(functions, targets):
```

Comment:
This should be private and renamed to `_harmonize_and_check_functions_and_targets`, i.e. there is a second and missing. 

Alternatively we could inline these steps into `create_dag` once we get rid of the redundant check at the beginning of `_create_concatenated_function_from_dag`.